pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -40,14 +40,36 @@ const (
 
 // Returns a configProcessingSkipFlag
 func parseConfig(config *zconfig.EdgeDevConfig, getconfigCtx *getconfigContext,
-	usingSaved bool) bool {
+	source configSource) bool {
 
 	// Do not accept new commands from Local profile server while new config
 	// from the controller is being applied.
-	getconfigCtx.localCommands.Lock()
-	defer getconfigCtx.localCommands.Unlock()
+	if getconfigCtx.localCommands != nil {
+		getconfigCtx.localCommands.Lock()
+		defer getconfigCtx.localCommands.Unlock()
+	}
+
+	// Update lastReceivedConfig even if the config processing is skipped.
+	if config.ConfigTimestamp.IsValid() {
+		getconfigCtx.lastReceivedConfig = config.ConfigTimestamp.AsTime()
+	} else {
+		getconfigCtx.lastReceivedConfig = time.Now()
+	}
+
+	// Make sure we do not accidentally revert to an older configuration.
+	// This depends on the controller attaching config timestamp.
+	// If not provided, the check is skipped.
+	if config.ConfigTimestamp.IsValid() {
+		configTimestamp := config.ConfigTimestamp.AsTime()
+		if getconfigCtx.lastConfigTimestamp.After(configTimestamp) {
+			log.Warnf("Skipping obsolete device configuration "+
```

Comment:
Can you add the config source to what is logged?

Reply:
done